### ISSUE TITLE
[BUG] Literal 'pthread' in .pc Libs.private

### ISSUE THREAD

**Issue by heftig** [Haivision/srt#1002]
Between 1.4.0 and 1.4.1, the pthread argument in the pkg-config files' `Libs.private` line was changed from `-lpthread` to `pthread`.

AFAIK (Arch Linux does not use static linking for SRT, so this line remains unused) this will cause a compiler error, as it is interpreted as an input file and not an option. The argument has to be either `-lpthread` or `-pthread`.

**Comment by ethouris** [comment]
Here is where this value comes from:

1. srt.pc  <-  scripts/srt.pc.in  : `Libs.private: @SRT_LIBS_PRIVATE@`
2. `CMakeLists.txt` file: `SRT_LIBS_PRIVATE` is updated in many places, here it is:
```
set (SRT_LIBS_PRIVATE ${SRT_LIBS_PRIVATE} ${PTHREAD_LIBRARY})
```
3. where `PTHREAD_LIBRARY` is set in the block starting with line 458, can be either provided by user by setting this variable, or set automatically:

On Windows:
```
		find_library(PTHREAD_LIBRARY NAMES ${PTHREAD_W32_LIBRARY} pthread pthread_dll pthread_lib HINTS C:/pthread-win32/lib C:/pthread-win64/lib ${PTHREAD_PACKAGE_LIB_HINT})
```
and on POSIX:
```
	set(PTHREAD_LIBRARY ${CMAKE_THREAD_LIBS_INIT})
```

CMake documentation doesn't say exactly what form the value of `CMAKE_THREAD_LIBS_INIT` has, but as I tested on the latest version in master (Linux, SuSE Leap), it prints `-lpthread` here:
```
Name: srt
Description: SRT library set
Version: 1.4.1
Libs: -L${libdir} -lsrt   
Libs.private:  -lpthread -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc
Cflags: -I${includedir} -I${includedir}/srt
Requires.private: openssl libcrypto
```

If you find something else then, it would be nice if you can trace the value in the cmake file - it must be some unique problem on this platform.

**Comment by heftig** [comment]
Sorry; this was a CMake bug.

Introduced in 3.16.0 and [fixed in 3.16.1](https://gitlab.kitware.com/cmake/cmake/commit/158839f0db905d44a4dd7d37e7bb36c1899fd17c).
